src/credentials/FabricTable.h

```diff
@@ -919,6 +919,12 @@ class DLL_EXPORT FabricTable
                                  FabricId & outFabricId, NodeId & outNodeId, Crypto::P256PublicKey & outNocPubkey,
                                  Crypto::P256PublicKey * outRootPublicKey = nullptr) const;
 
+    // Verifies credentials, using the provided root certificate.
+    // This call is done whenever a fabric is "directly" added
```

Comment:
This comment is no longer true.

The comments here need a _lot_ better documenation about why these two functions both exist.  And presumably the version that takes a fabric index needs to have so "stack locked by current thread" asserts?

Reply:
I don't think we should generally document who's calling which function, code search is the answer to that. The function should instead document what it does and any non-obvious details. I can see about improving some of this but honestly I found it in need of cleanup before I came in.

Reply:
Do we generally have "stack locked by current thread" asserts? Is there an example in CHIP code? I'm searching for calls to IsChipStackLockedByCurrentThread and I don't really see any.

Reply:
`assertChipStackLockedByCurrentThread`

Reply:
Done.